crates/solver/src/liquidity/order_converter.rs

```diff
@@ -58,12 +54,49 @@ impl OrderConverter {
             }
             OrderClass::Limit(_) => LimitOrderId::Limit(order.metadata.uid),
         };
+
+        // The reported fee amount that is used for objective computation is the
+        // order's full full amount scaled by a constant factor.
+        let mut solver_fee = remaining.remaining(order.metadata.solver_fee)?;
+        let mut sell_amount = remaining.remaining(sell_amount)?;
+        let mut buy_amount = remaining.remaining(order.data.buy_amount)?;
+
+        // Partially fillable orders are included in the auction when there is at least
+        // 1 atom balance available.
+        if order.data.partially_fillable {
+            let need = sell_amount
+                .checked_add(remaining.remaining(order.data.fee_amount)?)
+                .context("partially fillable")?;
+            let have = order
+                .metadata
+                .partially_fillable_balance
+                .context("no balance for partially fillable order")?;
+            tracing::trace!(%need, %have, "partially fillable order conversion");
+            if have < need {
+                solver_fee = solver_fee
+                    .checked_mul(have)
```

Comment:
You could create another `Remaining` instance from `have` and `need` to compute the adjusted amounts. Might look a little nicer.

Reply:
The Remaining struct does a similar calculation but it was made for the order executed amounts and for example numberator, denominator are private. I didn't want to generalise it for this so I repeated the calculation.